src/main/kotlin/com/cognifide/gradle/aem/instance/tasks/InstanceRestore.kt

```diff
@@ -0,0 +1,121 @@
+package com.cognifide.gradle.aem.instance.tasks
+
+import com.cognifide.gradle.aem.common.AemException
+import com.cognifide.gradle.aem.common.AemTask
+import com.cognifide.gradle.aem.common.Formats
+import com.cognifide.gradle.aem.common.file.FileOperations
+import com.cognifide.gradle.aem.common.file.resolver.FileResolver
+import com.cognifide.gradle.aem.common.onEachApply
+import com.cognifide.gradle.aem.common.tasks.LocalInstanceTask
+import com.cognifide.gradle.aem.instance.InstanceException
+import com.cognifide.gradle.aem.instance.LocalInstance
+import com.cognifide.gradle.aem.instance.names
+import com.fasterxml.jackson.annotation.JsonIgnore
+import java.io.File
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.TaskAction
+
+open class InstanceRestore : LocalInstanceTask() {
+
+    init {
```

Comment:
`init` block usually should be defined after fields definition.

if you will use some field and `init` block will be in place as yours (in the top) you will have NPE in Kotlin :D